packages/app/pages/index.tsx

```diff
@@ -100,13 +109,29 @@ const MembershipCard = (props: IMembershipCardProps) => {
           symbol: p[1],
           supply: p[2],
           price: web3.utils.fromWei(p[3], "ether"),
+          metadataHash: p[4],
         });
       });
     }
   }, [setPassport, passport, web3]);
   useEffect(() => {
     setUrl(props.redirect_url);
   }, [props.redirect_url]);
+  useEffect(() => {
+    if (!Object.entries(metadata).length && passport.metadataHash) {
+      axios
+        .post(
+          `https://ipfs.infura.io:5001/api/v0/block/get?arg=${passport.metadataHash}`,
+          {}
+        )
+        .then((r) => {
+          const jsonRegex = /{.*}/;
+          const metadataJson = jsonRegex.exec(r.data)?.[0] || "";
+          setMetadata(JSON.parse(metadataJson));
```

Comment:
This is the part I need the most help with.

It seems when I try to add a file to IPFS and then try to retrieve it, there are foreign characters added to the beginning and end of the data. I'm hacking around this with regexes to pull what I care about, but this seems bound to break eventually